internal/app/go-filecoin/node/block_propagate_test.go

```diff
@@ -108,6 +115,102 @@ func TestChainSync(t *testing.T) {
 	assert.True(t, equal, "failed to sync chains")
 }
 
+func TestChainSyncWithMessages(t *testing.T) {
+	tf.UnitTest(t)
+	ctx := context.Background()
+
+	/* setup */
+	// genesis has two accounts
+	genCfg := &gengen.GenesisCfg{}
+	require.NoError(t, gengen.MinerConfigs(MakeTestGenCfg(t, 1).Miners)(genCfg))
+	require.NoError(t, gengen.GenKeys(3)(genCfg))
+	require.NoError(t, gengen.GenKeyPrealloc(1, "100000")(genCfg))
+	require.NoError(t, gengen.GenKeyPrealloc(2, "100")(genCfg))
+	require.NoError(t, gengen.NetworkName(version.TEST)(genCfg))
+	cs := MakeChainSeed(t, genCfg)
+	fakeClock := th.NewFakeClock(time.Unix(1234567890, 0))
+	blockTime := 100 * time.Millisecond
```

Comment:
Given the use of a fake clock (:100), set a large block time like 30s so there's no chance that wall-clock time is actually involved.